### ISSUE TITLE
Does cluster-api renew certificates of workload cluster?

### ISSUE THREAD

**Issue by Bo0km4n** [kubernetes-sigs/cluster-api#5435]
**User Story**
In currently,  cluster-api generate certificates and embed these in the cloud-init script.
These expire in 1 year.
I expect cluster-api regenerate certificates before expire, and reflect these to file (e.g. `/etc/kubernetes/pki/apiserver.crt`) of workload cluster nodes.
Does cluster-api do it? 

As a [developer/user/operator] I would like to [high level description] for [reasons]

**Detailed Description**

[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**

[Miscellaneous information that will assist in solving the issue.]

/kind feature

**Comment by fabriziopandini** [comment]
In case you are using CABPK, CAPI leverages on kubeadm for creating certificates, and on top of my mind kubeadm doesn't allow to kustomize this value (but I might be wrong here). For other bootstrap providers things can be different

However, TBH I think that a better solution will be to have something (TBD with explicit opt in or opt out) that triggers CP rotation sometimes before certificates are about to expire.

**Comment by randomvariable** [comment]
On this specific issue, I think the options are:

a) Roll over KCP machines when a certain age is reached.
or
b) Open a proxy connection via the API server load balancer (i.e. the normal controlplane endpoint) to the API server port on the node, verify the time of the certificate, and make a decision on machine replacement based off of that.

**Comment by randomvariable** [comment]
We may want to revisit the milestone for this one @vincepri  @fabriziopandini .  I think it's more self-contained than #5490

**Comment by fabriziopandini** [comment]
/milestone v1.1
I agree this is a self-contained part of #5490, even if it will greatly benefit if we can get at least an high level feature of the end state before starting to execute, so we can ensure that all the efforts fit into a consistent story

**Comment by sbueringer** [comment]
@randomvariable Does b) then cover all relevant certificates? Or doe that not matter as they all have the same expiry?

**Comment by fabriziopandini** [comment]
/remove-lifecycle stale
@sbueringer to update/merge with other issue given I know he did some work in this area

**Comment by sbueringer** [comment]
Let's close this issue for now in favor of https://github.com/kubernetes-sigs/cluster-api/issues/5490

I'll update the issue hopefully soon and then probably create a separate issue for certificate renewal.

As of today the answer to the question in the issue title is no.

/close
